### PR TITLE
Load base

### PR DESCRIPTION
**Pull request by tjstienstra**
#### References to other Issues or PRs

Output of the mechanics experimental development (#24234)
Implements #24641

#### Brief description of what is fixed or changed
This PR implements load objects.

#### Release Notes

* physics.mechanics
  * Implemented `Force` and `Torque` objects to store the location and vector of the load.
  * BREAKING: moved the gravity function from `physics.mechanics.functions.py` to `physics.mechanics.loads.py`

### CODE REVIEW

File: doc/src/modules/physics/mechanics/masses.rst [hunk 1]
```diff
@@ -139,6 +139,38 @@ The mass is specified exactly as is in a particle. Similar to the
...
+   >>> Force(Po, N.x)
+   (Po, N.x)
+
+The location of a torque on the other had is a frame.
+
+   >>> from sympy.physics.mechanics import Torque
+   >>> Torque(N, 2 * N.x)
+   (N, 2*N.x)
+
+Optionally one can, when using the dedicated classes, also parse the body. If
```
Comment (brocksam) [inline doc/src/modules/physics/mechanics/masses.rst:163]:
```suggestion
Optionally one can also pass the body when using the dedicated classes. If
```

File: sympy/physics/mechanics/loads.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
+from abc import ABC
+from collections import namedtuple
+from sympy.physics.mechanics.body_base import BodyBase
+from sympy.physics.vector import Vector, ReferenceFrame, Point
+
+__all__ = ['LoadBase', 'Force', 'Torque']
+
+
+class LoadBase(ABC, namedtuple('LoadBase', ['location', 'vector'])):
```
Comment (brocksam) [inline sympy/physics/mechanics/loads.py:9]: Do we want `location` and `vector` as part of the "public" API given that we're giving both `Force` and `Torque` their own uniquely-named attributes for these? Could they be `_location` and `_vector` instead?

File: doc/src/modules/physics/mechanics/masses.rst [hunk 3]
```diff
@@ -139,6 +139,38 @@ The mass is specified exactly as is in a particle. Similar to the
...
+second argument is the vector. In the case of a force the first argument is a
+point and the second a vector.
+
+   >>> from sympy.physics.mechanics import Point, ReferenceFrame, Force
+   >>> N = ReferenceFrame('N')
+   >>> Po = Point('Po')
+   >>> Force(Po, N.x)
+   (Po, N.x)
+
+The location of a torque on the other had is a frame.
```
[the remark this answers is not in the document]
Comment (TJStienstra) [reply doc/src/modules/physics/mechanics/masses.rst:157]: Any better ideas? We can also change `location` to `reference`. The location a force acts upon is a line, and we use a `Point` as reference. For torque the location is generally a `RigidBody`, though we only care about the rotation, so a frame is our reference.

File: sympy/physics/mechanics/loads.py (hunk 2 above)
Comment (TJStienstra) [reply sympy/physics/mechanics/loads.py:9]: Can make the private. Reason to keep them public is to let people treat `Force` and `Torque` more similar.

File: sympy/physics/mechanics/tests/test_loads.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+from sympy.physics.mechanics.loads import Force, Torque, gravity, _parse_load
```
[the remark this answers is not in the document]
Comment (TJStienstra) [reply sympy/physics/mechanics/tests/test_loads.py:1]: Will do, had to import stuff from `.loads` anyways, but importing from `mechanics` will at least also test whether they are in the `__init__.py` file.

**Review by brocksam — APPROVED** [review]

File: sympy/physics/mechanics/loads.py (hunk 2 above)
Comment (TJStienstra) [reply sympy/physics/mechanics/loads.py:9]: Just found out another reason. `namedtuple` does not accept fields with leading underscores.

File: doc/src/modules/physics/mechanics/masses.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (smichr) [reply doc/src/modules/physics/mechanics/masses.rst]: > location a force acts upon is a line
To me this sounds like a direction (line) whereas location is a point or region of 3-D space (like a room).

Also, `had` -> `hand`

File: doc/src/modules/physics/mechanics/masses.rst [hunk 5]
```diff
@@ -139,6 +139,38 @@ The mass is specified exactly as is in a particle. Similar to the
...
+   >>> Force(Po, N.x)
+   (Po, N.x)
+
+The location of a torque on the other hand is a frame.
+
+   >>> from sympy.physics.mechanics import Torque
+   >>> Torque(N, 2 * N.x)
+   (N, 2*N.x)
+
+Optionally one can also pass the body when using the decicated classes. If so,
```
Comment (smichr) [inline doc/src/modules/physics/mechanics/masses.rst:163]:
```suggestion
Optionally, one can also pass the body when using dedicated classes. If so,
```

File: doc/src/modules/physics/mechanics/masses.rst [hunk 6]
```diff
@@ -139,6 +139,38 @@ The mass is specified exactly as is in a particle. Similar to the
...
 fashion and holds information about the body's orientation and angular velocity.
 
+Loads
+=====
+
+In :mod:`sympy.physics.mechanics` loads can either be represented with tuples or
+with the dedicated classes :class:`~.Force` and :class:`~.Torque`. Generally the
+first argument (or item in the case of a tuple) is the location of the load. The
+second argument is the vector. In the case of a force the first argument is a
+point and the second a vector.
```
Comment (smichr) [inline doc/src/modules/physics/mechanics/masses.rst:149]: It's perhaps way too late, but I wonder why mechanics didn't use geometric objects to do some of this, like using `Ray` for a force whose p1 is the point of application and the vector from p1 to p2 giving the direction and the magnitude, e.g. `force = lambda pt, mag, ang: Ray((0,0), (mag,0)).rotate(ang).translate(*pt)` for a force of magnitude 2 in the 45 deg direction from (1,2): `force((1,2),2,pi/4)`.

File: doc/src/modules/physics/mechanics/masses.rst [hunk 7]
```diff
@@ -139,6 +139,38 @@ The mass is specified exactly as is in a particle. Similar to the
...
+
+   >>> from sympy.physics.mechanics import Point, ReferenceFrame, Force
+   >>> N = ReferenceFrame('N')
+   >>> Po = Point('Po')
+   >>> Force(Po, N.x)
+   (Po, N.x)
+
+The location of a torque on the other hand is a frame.
```
Comment (smichr) [inline doc/src/modules/physics/mechanics/masses.rst:157]:
```suggestion
The location of a torque, on the other hand, is a frame.
```

File: doc/src/modules/physics/mechanics/masses.rst (hunk 6 above)
Comment (moorepants) [reply doc/src/modules/physics/mechanics/masses.rst:149]: Did that model exist back in 2010 or so? Also, everything in mechanics is based off the physics Vector which is how we manage these things.

Comment (moorepants) [reply doc/src/modules/physics/mechanics/masses.rst:149]: Looks like point.py might be as old as 2007. The answer is that we decided to implement a Vector object and we wrote the necessary methods for that object. I supposed we could have built Vector on top of some existing geometry tools back then, but it didn't happen.

**Comment by brocksam** [comment]
@TJStienstra can you please merge in `master` so that we can get the CI passing?

**Pull request merged**